### PR TITLE
add time sum reduction to dmUpdate

### PR DESCRIPTION
**Pull request by rem1776**
**Description**
Adds time_sum reduction method and adds a test based off the other time_* reduction tests.

I think the numbers from the added check test make sense, but let me know if you see any issues. I put some comments so its easier to tell what the results end up as.

**How Has This Been Tested?**
Tested on amd with oneapi 2023.2

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

**Review by uramirez8707 — APPROVED** [review]
This looks good to me

File: diag_manager/fms_diag_output_buffer.F90 [hunk 1]
```diff
@@ -571,5 +572,39 @@ function do_time_max_wrapper(this, field_data, mask, is_masked, bounds_in, bound
...
+      class default
+        err_msg="do_time_sum_wrapper::the output buffer and the buffer send in are not of the same type (r8_kind)"
+      end select
+    type is (real(kind=r4_kind))
+      select type (field_data)
+      type is (real(kind=r4_kind))
+        call do_time_sum(output_buffer, field_data, mask, is_masked, bounds_in, bounds_out, &
+          real(missing_value, kind=r4_kind))
+      class default
+        err_msg="do_time_sum_wrapper::the output buffer and the buffer send in are not of the same type (r4_kind)"
```
[the remark this answers is not in the document]
Comment (rem1776) [reply diag_manager/fms_diag_output_buffer.F90:605]: by the time this gets to an actual error message it'll add the field information to the returned string

File: diag_manager/include/fms_diag_reduction_methods.inc [hunk 2]
```diff
@@ -201,4 +201,71 @@ subroutine DO_TIME_MAX_ (data_out, data_in, mask, is_masked, bounds_in, bounds_o
...
+subroutine DO_TIME_SUM_(data_out, data_in, mask, is_masked, bounds_in, bounds_out, missing_value )
+  real(FMS_TRM_KIND_),       intent(inout) :: data_out(:,:,:,:,:) !< output data
+  real(FMS_TRM_KIND_),       intent(in)    :: data_in(:,:,:,:)    !< data to update the buffer with
+  logical,                   intent(in)    :: mask(:,:,:,:)       !< mask
+  logical,                   intent(in)    :: is_masked           !< .True. if the field is using a mask
+  type(fmsDiagIbounds_type), intent(in)    :: bounds_in           !< indices indicating the correct portion
+                                                                  !! of the input buffer
+  type(fmsDiagIbounds_type), intent(in)    :: bounds_out          !< indices indicating the correct portion
+                                                                  !! of the output buffer
+  real(FMS_TRM_KIND_),       intent(in)    :: missing_value       !< Missing_value for data points that are masked
```
Comment (thomas-robinson) [inline diag_manager/include/fms_diag_reduction_methods.inc:218]: Isn't this a fill value?

File: diag_manager/include/fms_diag_reduction_methods.inc [hunk 3]
```diff
@@ -201,4 +201,71 @@ subroutine DO_TIME_MAX_ (data_out, data_in, mask, is_masked, bounds_in, bounds_o
...
+              data_out(is_out + i, js_out + j, ks_out + k, l + 1, 1) =           &
+                         data_out(is_out + i, js_out + j, ks_out + k, l + 1, 1)  &
+                       + data_in(is_in +i, js_in + j, ks_in + k, l + 1)
+            else
+              data_out(is_out + i, js_out + j, ks_out + k, l + 1, 1) = missing_value
+            endif
+          enddo
+        enddo
+      enddo
+    enddo
```
Comment (thomas-robinson) [inline diag_manager/include/fms_diag_reduction_methods.inc:258]: Can this be reduced to a `where` and `elsewhere` constuct?

File: diag_manager/include/fms_diag_reduction_methods.inc (hunk 2 above)
Comment (rem1776) [reply diag_manager/include/fms_diag_reduction_methods.inc:228]: The _FillValue and a missing_value for variables are both set to the same number when i look at the files produced, i just used the argument name from the other reductions.

**Review by thomas-robinson — APPROVED** [review]

File: diag_manager/include/fms_diag_reduction_methods.inc (hunk 3 above)
Comment (thomas-robinson) [reply diag_manager/include/fms_diag_reduction_methods.inc]: I hope this works

**Pull request merged**